contracts/WakuRlnRegistry.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import {WakuRln} from "./WakuRln.sol";
+import {IPoseidonHasher} from "rln-contract/PoseidonHasher.sol";
+import {Ownable} from "openzeppelin-contracts/contracts/access/Ownable.sol";
+import "forge-std/console.sol";
+
+error StorageAlreadyExists(address storageAddress);
+error NoStorageContractAvailable();
+error IncompatibleStorage();
+error IncompatibleStorageIndex();
+
+contract WakuRlnRegistry is Ownable {
+    uint16 public nextStorageIndex;
+    mapping(uint16 => address) public storages;
+
+    uint16 public usingStorageIndex = 0;
+
+    IPoseidonHasher public immutable poseidonHasher;
+
+    event NewStorageContract(uint16 index, address storageAddress);
+
+    constructor(address _poseidonHasher) Ownable() {
+        poseidonHasher = IPoseidonHasher(_poseidonHasher);
+    }
+
+    function _insertIntoStorageMap(address storageAddress) internal {
+        storages[nextStorageIndex] = storageAddress;
+        emit NewStorageContract(nextStorageIndex, storageAddress);
+        nextStorageIndex += 1;
+    }
+
+    function registerStorage(address storageAddress) external onlyOwner {
+        if (storages[nextStorageIndex] != address(0)) revert StorageAlreadyExists(storageAddress);
+        WakuRln wakuRln = WakuRln(storageAddress);
+        if (wakuRln.poseidonHasher() != poseidonHasher) revert IncompatibleStorage();
+        if (wakuRln.contractIndex() != nextStorageIndex) revert IncompatibleStorageIndex();
+        _insertIntoStorageMap(storageAddress);
+    }
+
+    function newStorage() external onlyOwner {
+        WakuRln newStorageContract = new WakuRln(address(poseidonHasher), nextStorageIndex);
+        _insertIntoStorageMap(address(newStorageContract));
+    }
+
+    function register(uint256[] calldata commitments) external payable {
+        if (usingStorageIndex >= nextStorageIndex) revert NoStorageContractAvailable();
+
+        // iteratively check if the storage contract is full, and increment the usingStorageIndex if it is
+        while (true) {
+            try WakuRln(storages[usingStorageIndex]).register(commitments) {
```

Comment:
I wonder if we should ensure in `WakuRln._validateRegistration()` that a given `commitment` is a valid value.
What happens if an account calls `register()` with a bunch of `0` values? Can this happen? If yes, what impact would that have?

Reply:
Correct, we should add a check that the provided commitment is within the bn254 field - adding a check for zero values is a good idea, i'll upstream it to the RlnBase contract too! thanks

Reply:
Tracking https://github.com/vacp2p/rln-contract/issues/25